

library(EnhancedVolcano)
library(RColorBrewer)
library(org.Mm.eg.db)
library(ggplot2)
library(dplyr)
#-----------Figure 3d ---------------------
#Example of  Volcano plots of differentially expressed genes comparing +/+ = ApcΔ/ΔSmad4+/+, Δ/Δ = ApcΔ/ΔSmad4Δ/Δ at 0h 
 
 #Prepare the data 
 res_05 <- read.csv("~/LFShrinkage_res_05 FL_0h _vs_ WT_0h.csv"), header= T, stringsAsFactors = F, sep = ",", row.names = 1)

 res_05 <- as.data.frame(res_05)
  
degenes<- as.data.frame(res_05) # assign the results to data frame degenes
dim(degenes)
# In the next two steps, assign symbol(common names) and ENTREZ id according to ENSEMBL id
degenes$symbol<- mapIds(org.Mm.eg.db,
                        keys= rownames(degenes),
                        column = "SYMBOL",
                        keytype = "SYMBOL",
                        multiVals = "first")
## 'select()' returned 1:many mapping between keys and columns
degenes$entrez<- mapIds(org.Mm.eg.db,
                        keys= rownames(degenes),
                        column = "ENTREZID",
                        keytype = "SYMBOL",
                        mutiVals= "first")
# In the next two steps, assign symbol(common names) and ENTREZ id according to ENSEMBL id

## 'select()' returned 1:many mapping between keys and columns
print(dim(degenes))
degenes<- degenes[is.na(degenes$symbol)== FALSE,] # Remove all the rows where no mapping 
degenes<- degenes[!duplicated(degenes$symbol),] # Remove all the rows that had duplicated
print(dim(degenes))

degenes_symbol<- degenes[is.na(degenes$symbol)== FALSE,]
dim(degenes_symbol)
degenes_symbol<- degenes_symbol[!duplicated(degenes_symbol$symbol),]
print(dim(degenes_symbol))
degenes05<- subset(degenes_symbol, padj< 0.05) # subset the genes that have a padj< 0.05
#####Plot
EnhancedVolcano(res_05,
  lab = rownames(res_05),
  x = 'log2FoldChange',
  y = 'padj', 
 #y = 'pvalue',
  xlim = c(-6,6),
ylab = bquote(~-Log[10]~padj), 
 title = paste("Smad4",i),
  ylim=c(0, 30),
  xlab = bquote(~Log[2]~ 'fold change'),
  pCutoff =  1e-3,  # the default 10e-6
  FCcutoff = 1.5, # 
  pointSize = 2.0, #pointSize = c(ifelse(res2_05$log2FoldChange>2, 8, 1)), if you need to show the increase by changing size
  labSize = 5.0,
   shape = c(1, 4, 17, 18), #if we need to add shap
  col=c('black','gray','blue', 'red3'),
   colAlpha = 0.6,
  legendLabels = c('NS', expression(Log[2] ~ FC), expression(padj), 
      expression(padj ~ and ~ log[2] ~ FC)),
  legendPosition = 'bottom',
  legendLabSize = 10,
  legendIconSize = 3.0,
  drawConnectors = TRUE, #add arrows
  widthConnectors = 0.2,
  colConnectors = 'grey30',
    subtitle = "", #to remove Enhanched Volcano word
    selectLab = c("Id1", "Muc4", "Fn1",   "Id3", "Smad9", "Pla2g2a", "Sox2","Fyn",
            "Tgfb1", "Tgfb2", "Fos", "Skil",  "Ldlrad4", "Rgcc", "Serpine1","Itgb7", "Bmp6")) # selct gene names  upreguated in WT 0h vs Fl 0h for plotting

#########################################
#Pathway enrichment analysis using tmod with Hallmark 

#-----------Figure 3e---------------------


#Reference: https://d3amtssd1tejdt.cloudfront.net/2016/2420/1/tmod.pdf
#devtools::install_version("tmod", version = "0.40", repos = "http://cran.us.r-project.org")

## http://www.broadinstitute.org/gsea/downloads.jsp 
#download the  msigdb_v7.2.xml
#Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying overlaps between gene sets in other MSigDB collections and retaining genes that display coordinate expression.

library(tmod)
library(dplyr)
#read the files as follow
#The files are from Deseq2 output. They represent all All DEG after Shrinkage estimate without padj cuttof
FL_0h_vs_WT_0h <- read.csv("~/DEGs_All_LFShrinkage FL_0h _vs_ WT_0h .csv", sep=",")
FL_1h_vs_WT_1h <- read.csv("~/DEGs_All_LFShrinkage FL_1h _vs_ WT_1h .csv", sep=",")
FL_12h_vs_WT_12h <- read.csv("~/DEGs_All_LFShrinkage FL_12h _vs_ WT_12h .csv", sep=",")
WT_12h_vs_WT_0h <- read.csv("~/DEGs_All_LFShrinkage WT_12h _vs_ WT_0h .csv", sep=",")
WT_12h_vs_WT_1h <- read.csv("~/DEGs_All_LFShrinkage WT_12h _vs_ WT_1h .csv", sep=",")
WT_1h_vs_WT_0h <- read.csv("~/DEGs_All_LFShrinkage WT_1h _vs_ WT_0h .csv", sep=",")
Fl_12h_vs_Fl_0h <- read.csv("~/DEGs_All_LFShrinkage Fl_12h _vs_ Fl_0h .csv", sep=",")
Fl_12h_vs_Fl_1h <- read.csv("~/DEGs_All_LFShrinkage Fl_12h _vs_ Fl_1h .csv", sep=",")


#Read file that changes  mouse genes to human orthologs
Mous_hum <- read.table("~/MGI_to_HGNC_lookup.txt", sep="\t", header = T)


msig <- tmodImportMSigDB(
  file = "~/msigdb_v7.2.xml",
  format = "xml",
  organism = "Homo sapiens",
  fields = c("STANDARD_NAME", "CATEGORY_CODE", "SUB_CATEGORY_CODE",
             "EXTERNAL_DETAILS_URL") )

#We want to limit our analysis only to the 50 “hallmark”module categories

sel <-msig$MODULES$Category =="H" #H for hallmark 
table(msig$MODULES$Subcategory)
table(sel)
#change the gene names from mouse to human for all the files as follow

names(FL_0h_vs_WT_0h)[names(FL_0h_vs_WT_0h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
FL_0h_vs_WT_0h <- inner_join(Mous_hum ,FL_0h_vs_WT_0h, by = "Mouse.MGI" )
names(FL_1h_vs_WT_1h)[names(FL_1h_vs_WT_1h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
FL_1h_vs_WT_1h <- inner_join(Mous_hum ,FL_1h_vs_WT_1h, by = "Mouse.MGI" )
names(FL_12h_vs_WT_12h)[names(FL_12h_vs_WT_12h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
FL_12h_vs_WT_12h <- inner_join(Mous_hum ,FL_12h_vs_WT_12h, by = "Mouse.MGI" )
names(WT_12h_vs_WT_0h)[names(WT_12h_vs_WT_0h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
WT_12h_vs_WT_0h <- inner_join(Mous_hum ,WT_12h_vs_WT_0h, by = "Mouse.MGI" )
names(WT_12h_vs_WT_1h)[names(WT_12h_vs_WT_1h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
WT_12h_vs_WT_1h <- inner_join(Mous_hum ,WT_12h_vs_WT_1h, by = "Mouse.MGI" )
names(WT_1h_vs_WT_0h)[names(WT_1h_vs_WT_0h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
WT_1h_vs_WT_0h <- inner_join(Mous_hum ,WT_1h_vs_WT_0h, by = "Mouse.MGI" )
names(Fl_12h_vs_Fl_0h)[names(Fl_12h_vs_Fl_0h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
Fl_12h_vs_Fl_0h <- inner_join(Mous_hum ,Fl_12h_vs_Fl_0h, by = "Mouse.MGI" )
names(Fl_12h_vs_Fl_1h)[names(Fl_12h_vs_Fl_1h)=="symbol"] <-"Mouse.MGI" # change mouse genes to human ortholog 
Fl_12h_vs_Fl_1h <- inner_join(Mous_hum ,Fl_12h_vs_Fl_1h, by = "Mouse.MGI" )


#order the genes 
FL_0h_vs_WT_0h_genes <- FL_0h_vs_WT_0h[order(FL_0h_vs_WT_0h$padj),]
FL_1h_vs_WT_1h_genes <- FL_1h_vs_WT_1h[order(FL_1h_vs_WT_1h$padj),]
FL_12h_vs_WT_12h_genes <- FL_12h_vs_WT_12h[order(FL_12h_vs_WT_12h$padj),]
WT_12h_vs_WT_0h_genes <- WT_12h_vs_WT_0h[order(WT_12h_vs_WT_0h$padj),]
WT_12h_vs_WT_1h_genes <- WT_12h_vs_WT_1h[order(WT_12h_vs_WT_1h$padj),]
WT_1h_vs_WT_0h_genes <- WT_1h_vs_WT_0h[order(WT_1h_vs_WT_0h$padj),]
Fl_12h_vs_Fl_0h_genes <- Fl_12h_vs_Fl_0h[order(Fl_12h_vs_Fl_0h$padj),]
Fl_12h_vs_Fl_1h_genes <- Fl_12h_vs_Fl_1h[order(Fl_12h_vs_Fl_1h$padj),]

#run the rest
res.FL_0h_vs_WT_0h_genes <- tmodCERNOtest(FL_0h_vs_WT_0h_genes$Human.HGNC, mset=msig[sel]) #
res.FL_1h_vs_WT_1h_genes <- tmodCERNOtest(FL_1h_vs_WT_1h_genes$Human.HGNC, mset= msig[sel])
res.FL_12h_vs_WT_12h_genes <- tmodCERNOtest(FL_12h_vs_WT_12h_genes$Human.HGNC, mset= msig[sel])
res.WT_12h_vs_WT_0h_genes <- tmodCERNOtest(WT_12h_vs_WT_0h_genes$Human.HGNC, mset= msig[sel])
res.WT_12h_vs_WT_1h_genes <- tmodCERNOtest(WT_12h_vs_WT_1h_genes$Human.HGNC, mset= msig[sel])
res.WT_1h_vs_WT_0h_genes <- tmodCERNOtest(WT_1h_vs_WT_0h_genes$Human.HGNC, mset= msig[sel])
res.Fl_12h_vs_Fl_0h_genes <- tmodCERNOtest(Fl_12h_vs_Fl_0h_genes$Human.HGNC, mset= msig[sel])
res.Fl_12h_vs_Fl_1h_genes <- tmodCERNOtest(Fl_12h_vs_Fl_1h_genes$Human.HGNC, mset= msig[sel])


#Plot
#this plot is based on selected modules

tmodPanelPlot(list(FL_0h_vs_WT_0h=res.FL_0h_vs_WT_0h_genes,
                   FL_1h_vs_WT_1h=res.FL_1h_vs_WT_1h_genes,
                   FL_12h_vs_WT_12h=res.FL_12h_vs_WT_12h_genes,
                   WT_1h_vs_WT_0h=res.WT_1h_vs_WT_0h_genes,
                   WT_12h_vs_WT_0h=res.WT_12h_vs_WT_0h_genes,
                   WT_12h_vs_WT_1h=res.WT_12h_vs_WT_1h_genes,
                   Fl_12h_vs_Fl_0h=res.Fl_12h_vs_Fl_0h_genes,
                   Fl_12h_vs_Fl_1h=res.Fl_12h_vs_Fl_1h_genes),
              filter.rows.auc = 0.5,
              pie=pies, pie.style = "rug", grid="b", pval.thr = 10^-2, pval.thr.lower = 10^-6, 
              text.cex=1.2, filter.by.id = c("M5890", "M5902", "M5932" ,"M5896", "M5891", "M5897" ,"M5939" ,"M5941" ,"M5947" ,"M5930" ,"M5936",
                                             "M5942" ,"M5953" ,"M5924" ,"M5950" ,"M5906" ,"M5934" ,"M5898" ,"M5901" ,"M5907" ,"M5926",
                                             "M5913" ,"M5925"  ,"M5915" ,"M5946" ,"M5911" ,"M5921" ,"M5892" ,"M5903" ,"M5895"  ,"M5922"  ,"M5893" ,"M5916" ,"M5923" ,"M5944",
                                             "M5948" ,"M5908" ))